### PR TITLE
Expose setFlushOn function

### PR DESCRIPTION
**Pull request by rzhao271**
Add a `setFlushOn` function to help with https://github.com/microsoft/vscode/issues/123856

### CODE REVIEW

File: src/logger.cc [hunk 1]
```diff
@@ -50,6 +50,41 @@ NAN_METHOD(setLevel) {
...
 }
 
+NAN_METHOD(setFlushOn) {
+  if (!info[0]->IsNumber()) {
+    return Nan::ThrowError(Nan::Error("Provide flush level"));
+  }
+
+  const int64_t levelNumber = Nan::To<int64_t>(info[0]).FromJust();
+  spdlog::level::level_enum level;
+  switch (levelNumber) {
```
Comment (rzhao271) [inline src/logger.cc:60]: I would move this switch block into a separate function, but I'm not sure how to handle it returning both a `spdlog::level::level_enum` or a Nan Error.

Comment (deepak1556) [reply src/logger.cc:60]: The switch block can be replaced with

```c++
auto level = static_cast<spdlog::level::level_enum>(levelNumber);
if (level >= spdlog::level::n_levels) {
  return Nan::ThrowError(Nan::Error("Invalid level"));
}
```

also change it in other places where this switch block is repeated.

**Review by deepak1556 — COMMENTED** [review]
LGTM with minor style change

File: src/logger.cc (hunk 1 above)
Comment (rzhao271) [reply src/logger.cc]: I changed it around a bit. It seems that directly casting to an enum like that results in version-dependent behaviour: https://stackoverflow.com/questions/18195312/what-happens-if-you-static-cast-invalid-value-to-enum-class.

File: src/logger.cc [hunk 2]
```diff
@@ -20,36 +20,29 @@ NAN_METHOD(setLevel) {
...
   spdlog::set_level(level);
 }
 
+NAN_METHOD(setFlushOn) {
+  if (!info[0]->IsNumber()) {
+    return Nan::ThrowError(Nan::Error("Provide flush level"));
+  }
+
+  const int64_t levelNumber = Nan::To<int64_t>(info[0]).FromJust();
+  if (levelNumber >= spdlog::level::n_levels || levelNumber < 0) {
```
Comment (deepak1556) [inline src/logger.cc:38]:
```suggestion
  if (levelNumber >= spdlog::level::n_levels || levelNumber < spdlog::level::trace) {
```

this allows to catch any change in enum constants.

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**